scm-packaging/docker/build.gradle

```diff
@@ -70,45 +69,76 @@ task context(type: Copy) {
   }
 }
 
-task dockerImage(type: DockerBuildImage) {
-  inputDir = file('build/docker')
-  images = images()
+task setupBuilder() {
+  doLast {
+    def inspect = exec {
+      commandLine = ["docker", "buildx", "inspect", "scm-builder"]
+      ignoreExitValue = true
+    }
+    if (inspect.exitValue != 0) {
+      exec {
+        commandLine = ["docker", "run", "--privileged", "--rm", "tonistiigi/binfmt", "--install", "arm,arm64"]
+      }
+      exec {
+        commandLine = ["docker", "buildx", "create", "--name", "scm-builder", "--driver", "docker-container", "--platform", "linux/arm/v7,linux/arm64/v8,linux/amd64"]
+      }
+      exec {
+        commandLine = ["docker", "buildx", "inspect", "scm-builder"]
+      }
+    }
+  }
+}
+
+task build(type: Exec) {
+  commandLine = ["docker", "buildx", "bake", "--builder", "scm-builder", isSnapshot ? "dev": "prod"]
+  environment "VERSION", dockerTag
+  environment "COMMIT_SHA", revision
+  environment "IMAGE", dockerRepository
+
   doLast {
     File file = new File(project.buildDir, 'docker.tag')
     file.text = dockerTag
   }
-  dependsOn 'context'
+  dependsOn 'context', 'setupBuilder'
 }
 
-def images() {
-  if (isSnapshot) {
-    return [
-      "${dockerRepository}:${dockerTag}"
-    ]
-  } else {
-    // What about patch releases?
-    // It is a good idea to push always latest
-    return [
-      "${dockerRepository}:${dockerTag}",
-      "${dockerRepository}:latest"
-    ]
-  }
+task pushImages(type: Exec) {
+  commandLine = ["docker", "buildx", "bake", "--builder", "scm-builder",  isSnapshot ? "dev": "prod", "--push"]
+  environment "VERSION", dockerTag
+  environment "COMMIT_SHA", revision
+  environment "IMAGE", dockerRepository
```

Comment:
I would like to talk about how this docker-bake stuff work. Maybe even tell the whole team about what you build here?